src/resmom/linux/alps.c

```diff
@@ -3404,6 +3514,123 @@ disallow_char_data(ud_t *d, const XML_Char *s, int len)
 	return;
 }
 
+/**
+ * @brief
+ * Helper function for allow_char_data() that is registered to handle
+ * character data for elements.
+ * The user data structure (ud_t) is populated with the node rangelist associated
+ * with the 'Nodes' element (in the System BASIL 1.7 XML response), that is
+ * currently being processed.
+ *
+ * @param d pointer to user data structure.
+ * @param[in] s string.
+ * @param[in] len length of string.
+ *
+ * @return void
+ */
+static void
+parse_nidlist_char_data(ud_t *d, const char *s, int len)
+{
+	char *tmp_buffer, *tmp_ptr;
+
+	tmp_buffer = malloc(sizeof(char) * (len + 1));
+	if (tmp_buffer == NULL) {
+		parse_err_out_of_memory(d);
+		d->current_sys.node_group->nidlist = NULL;
+		return;
+	}
+
+	/* 's' is not guaranteed to be Null terminated. */
+	snprintf(tmp_buffer, len + 1, "%s", s);
```

Comment:
This will not require if we use strndup().

Reply:
Done!